### ISSUE TITLE
Horrible death results in unusable mount point. 

### ISSUE THREAD

**Issue by mlongval** [jstaf/onedriver#88]
I don't know if I am doing something wrong. 

OneDriver mounted to "OneDrive" mount point in my home works great until I suspend 
the machine for a certain length of time. 

Upon recovery from suspend, the onedriver process seems to die a horrible horrible death 
and leaves a weird  ????  ???? OneDrive  mount point in my directory. 
I cannot delete this directory and it keeps telling me that the "Transport has no end point"
The only way to recover is by restarting(rebooting) the computer. 
I tried looking for the onedriver process or something else that I could kill to allow deletion of the 
mountpoint but to no avail.

**Comment by jstaf** [comment]
`fusermount -uz MOUNTPOINT_HERE` will kill it and cleanup the mountpoint for you.

Alternatively, you can kill it via systemd, which just runs the command for you (if you know the service name):

```
# this is the command to cleanup the default mountpoint if you launched it via the app launcher/.desktop launcher

# do not run with sudo
systemctl --user stop onedriver@-home-$(whoami)-OneDrive.service
```

Out of curiosity, what's the best way of reproducing this? Suspend/hibernate the machine? If I can reproduce this I'll be able to patch out it in the next release.

**Comment by mlongval** [comment]
So far I've had it just by letting the systems sit. The first time was on a DigitalOcean droplet with Ubuntu 20.04. I just came back the next day and it was like that. Thank you very much for your help.

**Comment by jstaf** [comment]
I haven't been able to reproduce this yet, but I think what's happening is that onedriver is crashing after resuming from suspend and the log should indicate where in the code this is happening.

You can get the log from journalctl with `journalctl --user -u $SERVICE_NAME` (see https://github.com/jstaf/onedriver#multiple-drives-and-starting-onedrive-on-login for what value to use for $SERVICE_NAME). The type of crash you're seeing should have a pretty severe error similar to this in the log: `panic: runtime error: invalid memory address or nil pointer dereference`.

(I can probably see if I can add cleaner handling of these types of crashes in the next release.)

**Comment by jstaf** [comment]
I think I caught the error, will fix in next release:

```
Jul 13 11:07:26 onedriver[2345]: time="2020-07-13T11:07:26" level=error msg="Could not POST to renew tokens, forcing reauth." func="000099:fs/graph.(*Auth).Refresh()" file="oauth2.go:72" err="Post https://login.microsoftonline.com/common/oauth2/v2.0/token: dial tcp: lookup login.microsoftonline.com: Temporary failure in name resolution"
Jul 13 11:07:26 onedriver[2345]: panic: runtime error: invalid memory address or nil pointer dereference
Jul 13 11:07:26 onedriver[2345]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x6c6624]
Jul 13 11:07:26 onedriver[2345]: goroutine 99 [running]:
Jul 13 11:07:26 onedriver[2345]: github.com/jstaf/onedriver/fs/graph.(*Auth).Refresh(0xc000012400)
Jul 13 11:07:26 onedriver[2345]:         /build/onedriver-MYz9eI/onedriver-0.9.1/fs/graph/oauth2.go:78 +0x4d4
Jul 13 11:07:26 onedriver[2345]: github.com/jstaf/onedriver/fs/graph.Request(0xc00014e560, 0x98, 0xc000012400, 0x7cdd6d, 0x3, 0x0, 0x0, 0x4, 0x7e9ca0, 0x1, ...)
Jul 13 11:07:26 onedriver[2345]:         /build/onedriver-MYz9eI/onedriver-0.9.1/fs/graph/graph.go:42 +0x2d0
Jul 13 11:07:26 onedriver[2345]: github.com/jstaf/onedriver/fs/graph.Get(...)
Jul 13 11:07:26 onedriver[2345]:         /build/onedriver-MYz9eI/onedriver-0.9.1/fs/graph/graph.go:102
Jul 13 11:07:26 onedriver[2345]: github.com/jstaf/onedriver/fs.(*Cache).pollDeltas(0xc0000f21b0, 0xc000012400, 0xc000389e30, 0x1, 0x1, 0x1, 0x1, 0x0)
Jul 13 11:07:26 onedriver[2345]:         /build/onedriver-MYz9eI/onedriver-0.9.1/fs/delta.go:101 +0x77
Jul 13 11:07:26 onedriver[2345]: github.com/jstaf/onedriver/fs.(*Cache).DeltaLoop(0xc0000f21b0, 0x6fc23ac00)
Jul 13 11:07:26 onedriver[2345]:         /build/onedriver-MYz9eI/onedriver-0.9.1/fs/delta.go:25 +0x14d
Jul 13 11:07:26 onedriver[2345]: created by main.main
Jul 13 11:07:26 onedriver[2345]:         /build/onedriver-MYz9eI/onedriver-0.9.1/cmd/onedriver/main.go:119 +0x7bc
```

**Comment by jstaf** [comment]
This should be fixed in #91. 

Will see if I can add better segfault handling in a future release (so it avoids that "transport: endpoint is not connected error").
